src/discordas.ts

```diff
@@ -214,20 +216,67 @@ async function run(): Promise<void> {
 
     roomhandler.bindThirdparty();
 
+    try {
+        await startDiscordBot(discordbot, appservice.botClient, config.bridge.adminMxid);
+        log.info("Discordbot started successfully");
+    } catch (err) {
+        log.error(err);
+        log.error("Failure during startup. Exiting");
+        process.exit(1);
+    }
+
     await appservice.begin();
     log.info(`Started listening on port ${port}`);
 
+}
+
+interface AdminRooms {
+    [mxid: string]: string;
+}
+
+async function notifyBridgeAdmin(client: MatrixClient, adminMxid: string, message: string) {
```

Comment:
Is there a chance we need to avoid this being called rapidly and creating lots of rooms?

Reply:
Hmm, it will keep spamming rooms if it fails to invite the admin user (due to invalid MXID for example). Perhaps it should `try` it all and clean up the room if any of the steps after it fail.

Reply:
Should no longer do that as of 7b02c4f